test-data/unit/pythoneval.test

```diff
@@ -1924,3 +1924,49 @@ _testStarUnpackNestedUnderscore.py:10: error: List item 0 has incompatible type
 _testStarUnpackNestedUnderscore.py:10: error: List item 1 has incompatible type "int"; expected "str"
 _testStarUnpackNestedUnderscore.py:11: note: Revealed type is "builtins.list[builtins.str]"
 _testStarUnpackNestedUnderscore.py:16: note: Revealed type is "builtins.list[builtins.object]"
+
+[case testStrictEqualitywithParamSpec]
```

Comment:
I feel like this test case should ideally be in `check-expressions.test` rather than `pythoneval.test`. I tried putting this test case in `check-expressions.test`:

```
[case testStrictEqualitywithParamSpec]
# flags: --strict-equality
from typing import Generic
from typing_extensions import ParamSpec

P = ParamSpec("P")

class Foo(Generic[P]): ...

def check(foo1: Foo[[int]], foo2: Foo[[str]]) -> bool:
    return foo1 == foo2
[builtins fixtures/paramspec.pyi]
```

But I couldn't reproduce the crash with this test case in that file. (I tried a few modifications to various fixtures, still couldn't figure out how to reproduce the crash in the context of the test suite.) If anybody has any ideas there, I'd be interested to hear them.

Reply:
In a case like this it's fine to use a pythoneval test case. Even if we can figure out how to use test fixtures, there's a risk that the real stubs will diverge and will become problematic in the future.